src/search-manager/FilterByBlockType.tsx

```diff
@@ -57,41 +240,22 @@ const FilterByBlockType: React.FC<Record<never, never>> = () => {
     sortedBlockTypes[key] = blockTypes[key];
   });
 
-  const handleCheckboxChange = React.useCallback((e) => {
-    setBlockTypesFilter(currentFilters => {
-      if (currentFilters.includes(e.target.value)) {
-        return currentFilters.filter(x => x !== e.target.value);
-      }
-      return [...currentFilters, e.target.value];
-    });
-  }, [setBlockTypesFilter]);
-
   return (
     <SearchFilterWidget
       appliedFilters={blockTypesFilter.map(blockType => ({ label: <BlockTypeLabel type={blockType} /> }))}
       label={<FormattedMessage {...messages.blockTypeFilter} />}
-      clearFilter={() => setBlockTypesFilter([])}
+      clearFilter={clearFilters}
       icon={FilterList}
     >
       <Form.Group className="mb-0">
         <Form.CheckboxSet
           name="block-type-filter"
-          defaultValue={blockTypesFilter}
+          value={blockTypesFilter}
         >
           <Menu className="block-type-refinement-menu" style={{ boxShadow: 'none' }}>
             {
               Object.entries(sortedBlockTypes).map(([blockType, count]) => (
-                <label key={blockType} className="d-inline">
-                  <MenuItem
-                    as={Form.Checkbox}
-                    value={blockType}
-                    checked={blockTypesFilter.includes(blockType)}
-                    onChange={handleCheckboxChange}
-                  >
-                    <BlockTypeLabel type={blockType} />{' '}
-                    <Badge variant="light" pill>{count}</Badge>
-                  </MenuItem>
-                </label>
+                <FilterItem blockType={blockType} count={count} />
```

Comment:
I think we should consider having two separate components like `<FilterItem>` and `<ProblemFilterItem>`, because otherwise there is a lot of code in `FilterItem` that is only used in the case of `problem`

Reply:
Updated https://github.com/openedx/frontend-app-course-authoring/pull/1207/commits/f430c32e161ac1cf620d88a06b780f977e2e3b75